### ISSUE TITLE
Virtual terminal wrongly parsed input buffer under load

### ISSUE THREAD

**Issue by skyline75489** [microsoft/terminal#3073]
# Environment

```none
Windows build number: Version 10.0.18362.356
Windows Terminal version (if applicable): dev build on 0691c21876229dc573a18473ca04bfc78c0d6803

Any other software?
```

# Steps to reproduce

Open several `cmatrix` tabs and see what it looks like. Some color attribute are displayed as `32m`. Now open a new tab and quickly type things, the buffer are likely to be messed up.

# Expected behavior

Rendering is correct.

# Actual behavior

Not really.

# Analysis

As a follow-up of #1311, in which other people also noticed this problem. It seems like one tab is affecting another when under heavy load. I've explored the code a little bit and found that it was not the case. 

When under heavy load, contty is lagged and compressed several sequential key event as one. For example, if I quickly typed `aaaaaaslkd`, the received buffer in `StateMachine::ProcessString` ([link](https://github.com/microsoft/terminal/blob/master/src/terminal/parser/stateMachine.cpp#L1343)) looks like this:

```plaintext
"\x1b[93m\x1b[15;22Haaaaaa\x1b[?25h\x1b[m\x1b[?25l\x1b[93m\x1b[15;22Haaaaaas\x1b[?25h\x1b[m\x1b[?25l\x1b[93m\x1b[15;22Haaaaaasl\x1b[?25h\x1b[m\x1b[?25l\x1b[93m\x1b[15;22Haaaaaaslk\x1b[?25h\x1b[m\x1b[?25l\x1b[93m\x1b[15;22Haaaaaaslkd\x1b[?25h\x1b[m\x1b[?25l\x1b[93m\x1b[15;22Haaaaaas...
```

If we break it up, it wiil be:

```plaintext
\x1b[93m\x1b[15;22Haaaaaa\x1b[?25h\x1b[m\x1b[?25l
\x1b[93m\x1b[15;22Haaaaaas\x1b[?25h\x1b[m\x1b[?25l
\x1b[93m\x1b[15;22Haaaaaasl\x1b[?25h\x1b[m\x1b[?25l
\x1b[93m\x1b[15;22Haaaaaaslk\x1b[?25h\x1b[m\x1b[?25l
\x1b[93m\x1b[15;22Haaaaaaslkd\x1b[?25h\x1b[m\x1b[?25l
\x1b[93m\x1b[15;22Haaaaaas...
```

This seems reasonable. But `StateMachine::ProcessString` seems to be wrongly handling this and print control string like `\x1b[?25h` into the screen buffer. 

I've look to hear you guys opinion on this one, especially @zadjii-msft who left a comment in #1311.

**Comment by DHowett-MSFT** [comment]
So, there's a chance that this could be related to #3108 (fixed in #3110) -- the VT parsers were accidentally sharing global state, and that sharing would result in one VT parser "completing" parsing on another's behalf leaving the remaining one to just pop its remaining sequences out as text.

I've tried four tabs with cmatrix running as a test with #3110 in, and I can't reproduce this issue any longer.

**Comment by skyline75489** [comment]
@DHowett-MSFT That's probably it. I've tried cmatrix with #3110 and I can't reproduce this, either. Turns out it is tabs affecting each other, just in a very unusual and subtle way.
